### ISSUE TITLE
one should be able to re-join a room they left

### ISSUE THREAD

**Issue by AmandineLP** [vector-im/riot-web#3093]
not if they were kicked or banned but if they left voluntarily

**Comment by ara4n** [comment]
you can. @AmandineLP, what are the repro steps here?

**Comment by AmandineLP** [comment]
eg I left https://riot.im/develop/#/room/#help:matrix.org yesterday.

I went in Historical today, clicked on the room, clicked joined and got the following msg:
![image](https://cloud.githubusercontent.com/assets/8582858/22775704/1a4ac0f6-eead-11e6-977b-eabe69f92b42.png)

**Comment by AmandineLP** [comment]
Another example:
Create a room
Leave it
Click on it in Historical
Click on "click here"
This time I get:
![image](https://cloud.githubusercontent.com/assets/8582858/22778909/f23124b4-eeb8-11e6-9fea-89d11e8932c1.png)

**Comment by ara4n** [comment]
MMJD looks to have made #help:matrix.org invite-only.

The fact you can't rejoin empty rooms is a design flaw that's been in Matrix since day 1: https://github.com/matrix-org/matrix-doc/issues/534

**Comment by ara4n** [comment]
are you seeing this in any other rooms?

**Comment by AmandineLP** [comment]
No ok, works if the room is accessible by anyone having the link, in both including and excluding guests configurations.
I guess the problem is that by default every room is invite only. Surely if I've been invited once and I leave I should still be considered as invited if I want to rejoin. I'm especially thinking about people leaving by mistake. 

I agree this is more debatable than failing to rejoin any room, but I think this use case should be considered

**Comment by ara4n** [comment]
yeah, it's an open point as to whether invites should last forever or be once-off. we went for once-off in order to be a bit more security conscious, but up for debate....

**Comment by Matrixcoffee** [comment]
Should users who have invite power in a room be able to:
1. Join the room without an invite?
2. Invite themselves into the room, even when they aren't currently joined?

Strictly speaking, invite power only means (2), but security-wise (1) is the same as (2) with much better UX.

**Comment by kittykat** [comment]
> yeah, it's an open point as to whether invites should last forever or be once-off. we went for once-off in order to be a bit more security conscious, but up for debate....

Need product decision on ^ to close or proceed

**Comment by daniellekirkwood** [comment]
I've created a document that is currently awaiting a PR review on Element-Meta.

Invites should be used once only. If the user has been invited and accepted, then left (or removed) they will need a new invite to re-join the room. 

I believe that's how it works today. 

**If there's something I'm missing please comment here before May 6, 2022 else we'll close this issue with no action needed.**

**Comment by daniellekirkwood** [comment]
Closing; No action needed. See above comment.
